db/migrations/20201214160608_update_flip_bid_snapshot.sql

```diff
@@ -0,0 +1,149 @@
+-- +goose Up
+-- SQL in this section is executed when the migration is applied.
+
+-- add a flip_address attribute to the flip_bid_snapshot type
+ALTER TYPE api.flip_bid_snapshot ADD ATTRIBUTE flip_address text;
+
+-- update the get_flip function to return a flip_address in it's flip_bid_snapshot
+CREATE OR REPLACE FUNCTION api.get_flip(bid_id NUMERIC, ilk TEXT, block_height BIGINT DEFAULT api.max_block())
+    RETURNS api.flip_bid_snapshot
+AS
+$$
+WITH ilk_ids AS (SELECT id FROM maker.ilks WHERE ilks.identifier = get_flip.ilk),
+     -- there should only ever be 1 address for a given ilk, which is why there's a LIMIT with no ORDER BY
```

Comment:
Probably need to keep reading further down, but why wouldn't it be possible for there to be multiple addresses for a given ilk? Wouldn't like `MCD_FLIP_ETH_A_1_0_0` and `MCD_FLIP_ETH_A_1_0_9` both match on the `ilk` of `ETH_A` but exist as separate addresses on the `public.addresses` table?
Assuming that's the case, then I would think that the `address_id` we get back from `maker.flip_ilk` would be potentially non-deterministic

Reply:
Oops, I think that this comment is copy-pasted from the initial implementation of `get_flip` - I believe that I changed `get_flip` to `get_flip_with_address`, and everywhere that was calling `get_flip` should now be updated to use the new function. I did it this way, with several migrations, so that I could make the change iteratively and so it would hopefully be a bit easier to review. But this is a good reminder that it would probably best to squash all (most?) of these migrations into one before merging.

Reply:
Definitely on board with squashing. Also I suppose we're back in the realm of verifying that migrations timestamps are always only increasing from the most recent in the DB. Hopefully CI will save us there but probably worth keeping in the back of our minds as we're adding new ones

Reply:
That's a good call - I think that I ran into a failed build when the timestamps were out of order, so hopefully we'll be okay on that front.